File: Algorithm-Swift/Algorithm-Swift/Inflearn/09_모두의약수.swift
//
//  09_모두의약수.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/27.
//

import Foundation

func solution09() {
    let num = Int(readLine()!)!
    var counts = Array(repeating: 0, count: num)

    for i in 1...num {
        var cnt = 1;
        for j in stride(from: i, through: num, by: i * cnt) {
            counts[j-1] += 1
            cnt += 1
        }
    }

    counts.forEach { print($0, terminator: " ") }
}

File: Algorithm-CPP/09_모두의약수.cpp
#include <stdio.h>
using namespace std;

int main() {
    int num;
    scanf("%d", &num);

    int count[num];

    for(int i = 0; i<num; i++) {
        count[i] = 0;
    }

    for(int i = 0; i<num; i++) {
        for (int j = i; j<num; j++) {
            if((j + 1) % (i + 1) == 0) {
                count[j]++;
            }
        }
    }

    for(int i = 0; i<num; i++) {
        printf("%d ", count[i]);
    }

    return 0;
}